pkg/reconciler/managed/reconciler.go

```diff
@@ -1159,45 +1154,48 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 		log.Debug("External resource differs from desired state", "diff", observation.Diff)
 	}
 
-	// skip the update if the management policy is set to ignore updates
-	if !policy.ShouldUpdate() {
-		reconcileAfter := r.pollIntervalHook(managed, r.pollInterval)
-		log.Debug("Skipping update due to managementPolicies. Reconciliation succeeded", "requeue-after", time.Now().Add(reconcileAfter))
-		managed.SetConditions(xpv1.ReconcileSuccess())
-		return reconcile.Result{RequeueAfter: reconcileAfter}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
-	}
+	if !observation.ResourceUpToDate && policy.ShouldUpdate() {
+		update, err := external.Update(externalCtx, managed)
+		if err != nil {
+			// We'll hit this condition if we can't update our external resource,
+			// for example if our provider credentials don't have access to update
+			// it. If this is the first time we encounter this issue we'll be
+			// requeued implicitly when we update our status with the new error
+			// condition. If not, we requeue explicitly, which will trigger backoff.
+			log.Debug("Cannot update external resource")
+			record.Event(managed, event.Warning(reasonCannotUpdate, err))
+			managed.SetConditions(xpv1.Updating(), xpv1.ReconcileError(errors.Wrap(err, errReconcileUpdate)))
+			return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
```

Comment:
This is mostly the point of the PR. This is what gives us more consistent behaviour for `Update`.

Reply:
Also similar additions on line 1177, for example.